core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcherArguments.scala

```diff
@@ -44,7 +44,7 @@ private[mesos] class MesosClusterDispatcherArguments(args: Array[String], conf:
       port = value
       parse(tail)
 
-    case ("--webui-port" | "-p") :: IntParam(value) :: tail =>
```

Comment:
I agree that `-p` would always be matched by the pattern above, `"--port" | "-p"`. So removing it here wouldn't have any user-visible effects.